nexus/src/external_api/http_entrypoints.rs

```diff
@@ -5333,6 +5334,44 @@ async fn sled_view(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+// Physical disks
+
+/// List physical disks attached to sleds
+#[endpoint {
+    method = GET,
+    path = "/system/hardware/sleds/{sled_id}/disks",
+    tags = ["system"],
+}]
+async fn physical_disks_list(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    path_params: Path<SledPathParam>,
+    query_params: Query<PaginatedById>,
+) -> Result<HttpResponseOk<ResultsPage<PhysicalDisk>>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let query = query_params.into_inner();
```

Comment:
Something @davepacheco noted in another PR that I'm starting to do with all the API is to put everything inside the handler that can go there. These are fairly minor, but the handler itself is what's instrumented with a timer. If you see any of the API endpoints that have everything nested in the handler, that's why. You don't need to worry about doing that here, I'll catch it when I go over this for V1.

Reply:
Heh I was actually just following the pattern set by other entrypoints - which appear to parse inputs before calling "the main body of the endpoint" in a handler - but you're totally right, that distinction doesn't really make sense, we might as well instrument the whole thing.

Done in 3ff99699e27d669da1704562354a46a56c6df80d